### PR TITLE
Fix false positive in `yoda_condition` rule basing it on SwiftSyntax

### LINKED ISSUE #4081: yoda_condition false positive for bit field test
### New Issue Checklist

- [x] Updated SwiftLint to the latest version
- [x] I searched for [existing GitHub issues](https://github.com/realm/SwiftLint/issues)

### Describe the bug

When testing an integer bit field the yoda_condition rule triggers a false positive.

```swift
if flags & 1 == 1 {
    // ...
}

##### Complete output when running SwiftLint, including the stack trace and command used

```bash
$ swiftlint lint
file.swift:2:1: warning: Yoda condition rule Violation: The variable should be placed on the left, the constant on the right of a comparison operator. (yoda_condition)
```

### Environment

* SwiftLint version (run `swiftlint version` to be sure)? 0.47.1
* Installation method used (Homebrew, CocoaPods, building from source, etc)? brew
* Paste your configuration file: N/A
* Are you using [nested configurations](https://github.com/realm/SwiftLint#nested-configurations)? N/A
* Which Xcode version are you using (check `xcodebuild -version`)? N/A
* Do you have a sample that shows the issue?

```bash
> echo "if b & 1 == 1" | swiftlint lint --no-cache --use-stdin --enable-all-rules
<nopath>:1:1: warning: Yoda condition rule Violation: The variable should be placed on the left, the constant on the right of a comparison operator. (yoda_condition)
Done linting! Found 1 violation, 0 serious in 1 file.
```

### PR DESCRIPTION
**Pull request by SimplyDanny**
Fixes #4081.

### CODE REVIEW

**Comment by jpsim** [comment]
The bazel CI jobs were failing due to a compiler crash, they passed after a few retries. I'm not sure if it's a compiler issue or a rules_swift issue, but I filed https://github.com/bazelbuild/rules_swift/issues/875 to report since it might be an issue there.

Will review your changes shortly.

File: Source/SwiftLintFramework/Rules/Lint/YodaConditionRule.swift [hunk 1]
```diff
@@ -1,41 +1,15 @@
-import Foundation
 import SourceKittenFramework
+import SwiftSyntax
 
-public struct YodaConditionRule: ASTRule, OptInRule, ConfigurationProviderRule {
+public struct YodaConditionRule: OptInRule, ConfigurationProviderRule {
```
Comment (jpsim) [inline Source/SwiftLintFramework/Rules/Lint/YodaConditionRule.swift:4]: No longer makes any SourceKit requests:

```suggestion
public struct YodaConditionRule: OptInRule, ConfigurationProviderRule, SourceKitFreeRule {
```

**Review by jpsim — APPROVED** [review]
Nice work!

**Comment by Bouke** [comment]
Nice indeed!

**Pull request merged**